instance/migrations/0040_auto_20160420_0754.py

```diff
@@ -4,6 +4,27 @@
 from django.db import migrations, models
 
 
+def set_instance_states(apps, schema_editor):
+    SingleVMOpenEdXInstance = apps.get_model("instance", "SingleVMOpenEdXInstance")
+    for instance in SingleVMOpenEdXInstance.objects.iterator():
+        server_status = instance.server_status
+        if server_status:
+            if server_status.state_id in ("new", "started", "active", "rebooting"):
+                instance._status_to_waiting_for_server()
+            elif server_status.state_id in ("booted", "ready"):
+                instance._status_to_waiting_for_server()
+                instance._status_to_configuring_server()
+                instance._status_to_running()
+            elif server_status.state_id == "provisioning":
+                instance._status_to_waiting_for_server()
+                instance._status_to_configuring_server()
+            elif server_status.state_id == "terminated":
+                instance._status_to_waiting_for_server()
+                instance._status_to_configuring_server()
+                instance._status_to_running()
+                instance._status_to_terminated()
```

Comment:
A much more robust way to write this sort of migration would be:

```python
SingleVMOpenEdXInstance.objects.filter(server_set___status__in=("new", "started", "active", "rebooting")).update(_status=SingleVMOpenEdXInstance.Status.WaitingForServer.state_id)
SingleVMOpenEdXInstance.objects.filter(server_set___status__in=("booted", "ready")).update(_status=SingleVMOpenEdXInstance.Status.Running.state_id)
...
```

I believe that in the code now, `server_set` should only ever return one server. However, In the event that `server_set` contains multiple servers, the instance will be returned separately for each server, and this would potentially use the "wrong" server for the final status. Shouldn't happen though, so I'm not worried about that.